src/NHibernate/Multi/QueryBatch.cs

```diff
@@ -195,5 +203,110 @@ protected void ExecuteBatched()
 					stopWatch.Elapsed);
```

Comment:
I've basically copy pasted this statistic call from multi query/criteria implementation. But it seems it's not correct usage - as it expects query string as parameter and use this query string as dictionary key later. So it seems `resultSetsCommand.Sql` or some other unique batch identifier needs to be added here to `$"{_queries.Count} queries"` part.

Reply:
Something like `StringHelper.CollectionToString(_queries)` should do it, provided a suitable `ToString` override is added on query items. I would rather do that on the PR for handling cache mode.

Reply:
Well... Using `resultSetsCommand.Sql` sounds as much simpler solution. Considering that caching changes number of really executed queris per batch

Reply:
Putting SQL here is not an issue. But are this stats meant to cover only db hits or also cache hits?

Looking at their other usages, it is only db hits, so then better use `resultSetsCommand.Sql` than `_queries` without filtering those retrieved from cache. Note that we have the `QueryIdentifier`, meant to be provided to these stats, available in `CachingInformation`. But since these `CachingInformation` are optional, we cannot use that.